app/scripts/lib/createMetaRPCHandler.js

```diff
@@ -22,6 +28,15 @@ const createMetaRPCHandler = (api, outStream) => {
       result = await api[data.method](...data.params);
     } catch (err) {
       error = err;
+    } finally {
+      if (store && data.method !== 'getState') {
+        // we retrieve the already persisted data from local store to provide the version metadata to this persist call
+        const versionData = await localStore.get();
```

Comment:
We could move this, as well as the choice in store, into a module representing our storage layer. That would remove further duplicate code, and it would draw a clearer line between which code is responsible for persisting data, and which is responsible for sending RPC messages.

I understand the need for the RPC client to trigger data to be persisted - that makes sense. But it doesn't make sense to me for it to decide _where_ to persist it or how, and that's happening here. We should be able to capture this responsibility better with an abstraction.

Reply:
Thanks! So this seems like a good start. We can move further discussion to other threads and resolve this one.